### PR TITLE
[SDK-2798] Update Go Backend QS with go modules reference

### PR DESCRIPTION
**Pull request by sergiught**
This PR updates our Go backend QS to reference the use of go modules when adding dependencies and as well the go version required for this was bumped to go 1.17+.

## Reference PR: 

- https://github.com/auth0-samples/auth0-golang-api-samples/pull/38

## Screenshots:

image

image

### CODE REVIEW

File: articles/quickstart/backend/golang/index.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ github:
   org: auth0-samples
   repo: auth0-golang-api-samples
 requirements:
-  - Go 1.8.2
+  - Go 1.17+
```
Comment (Widcket) [inline articles/quickstart/backend/golang/index.yml]: Is this the absolute minimum version that is needed to run the QS sample app?

File: articles/quickstart/backend/golang/01-authorization.md [hunk 2]
```diff
@@ -23,15 +23,14 @@ useCase: quickstart
 
 ## Validate Access Tokens
 
-### Install dependencies
+### Add dependencies
+
+This example uses Go Modules and will download automatically all the needed dependencies during the build process.
 
 The [**form3tech-oss/jwt-go**](https://github.com/form3tech-oss/jwt-go) package can be used to verify incoming JWTs. The [**auth0/go-jwt-middleware**](https://github.com/auth0/go-jwt-middleware) library can be used alongside it to fetch your Auth0 public key and complete the verification process. Finally, we'll use the [**gorilla/mux**](https://github.com/gorilla/mux) package to handle our routes and [**codegangsta/negroni**](https://github.com/urfave/negroni) for HTTP middleware.
```
Comment (Widcket) [inline articles/quickstart/backend/golang/01-authorization.md:30]: Similarly to the samples PR, you could update the reference to the negroni and jwt-go repos. Though this could also be raised in a separate PR. Or in this PR, holding the merge until the other samples PRs get merged.

**Review by Widcket — COMMENTED** [review]
Looks good, just have a question regarding the Go version requirement.

File: articles/quickstart/backend/golang/01-authorization.md (hunk 2 above)
Comment (sergiught) [reply articles/quickstart/backend/golang/01-authorization.md:30]: Thanks for the suggestion @Widcket ! I would prefer to keep each PR isolated in scope and raise multiple ones so they're attached to different tickets and easier to track.

**Review by Widcket — APPROVED** [review]
